我下载的最新版的0.7.17，腾讯视频无法解析，而原版的you-get可以解析

D:\annie>annie -i https://v.qq.com/x/cover/2aya3ibdmft6vdw/e0765r4mwcr.html
panic: request error: %!s(<nil>) HTTP 403

goroutine 1 [running]:
github.com/iawia002/annie/request.Request(0x71eef5, 0x3, 0xc04225a000, 0x10c, 0x
0, 0x0, 0xc04218b5e8, 0xc000000073)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/request/request.go:
96 +0xc03
github.com/iawia002/annie/request.Headers(0xc04225a000, 0x10c, 0xc0421be180, 0x2
0, 0x10c)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/request/request.go:
150 +0x117
github.com/iawia002/annie/request.Size(0xc04225a000, 0x10c, 0xc0421be180, 0x20,
0x3)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/request/request.go:
156 +0x54
github.com/iawia002/annie/extractors.qqGenFormat(0xc04206c0a9, 0xb, 0xc0421be180
, 0x20, 0xc0421d00c0, 0x3, 0x4, 0xc042232340, 0x1, 0x4, ...)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/extractors/qq.go:10
1 +0x799
github.com/iawia002/annie/extractors.QQ(0xc04206c080, 0x39, 0x0, 0x0, 0x0, 0x0,
0x0, 0x0, 0x0, 0x0, ...)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/extractors/qq.go:14
0 +0x34a
main.download(0xc04206c080, 0x39)
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/main.go:97 +0x307
main.main()
        /Users/mac/iawia002/go/src/github.com/iawia002/annie/main.go:134 +0xe7

--- a/extractors/facebook_test.go
+++ b/extractors/facebook_test.go
@@ -18,7 +18,6 @@ func TestFacebook(t *testing.T) {
 			args: test.Args{
 				URL:     "https://www.facebook.com/JackyardsCovers/videos/vb.267832806658747/1215502888558396/",
 				Title:   "Jackyards - Great video, beautifully shot down Bournemouth...",
-				Size:    7105632,
 				Quality: "hd",
 			},
 		},

--- a/extractors/qq.go
+++ b/extractors/qq.go
@@ -53,36 +53,46 @@ func qqGenFormat(vid, cdn string, data qqVideoInfo) map[string]downloader.Format
 	var vkey string
 	// number of fragments
 	clips := data.Vl.Vi[0].Cl.Fc
-	fc := clips
 	if clips == 0 {
 		clips = 1
 	}
-	fns := strings.Split(data.Vl.Vi[0].Fn, ".")
 
-	if fc > 0 {
-		// If the number of fragments > 0, the filename needs to add the number of fragments
-		// n0687peq62x.p709.mp4 -> n0687peq62x.p709.1.mp4
-		fns = append(fns[:2], append([]string{"1"}, fns[2:]...)...)
-	}
-	var fmtIDPrefix string
 	for _, fi := range data.Fl.Fi {
-		if fc > 0 {
-			// Multiple formats
-			if fi.ID > 100000 {
-				fmtIDPrefix = "m"
-			} else {
-				fmtIDPrefix = "p"
-			}
-			fmtIDName := fmt.Sprintf("%s%d", fmtIDPrefix, fi.ID%10000)
-			fns[1] = fmtIDName
+		var fmtIDPrefix string
+		fns := strings.Split(data.Vl.Vi[0].Fn, ".")
+		if fi.ID > 100000 {
+			fmtIDPrefix = "m"
+		} else if fi.ID > 10000 {
+			fmtIDPrefix = "p"
 		}
+		if fmtIDPrefix != "" {
+			fmtIDName := fmt.Sprintf("%s%d", fmtIDPrefix, fi.ID%10000)
+			if len(fns) < 3 {
+				// v0739eolv38.mp4 -> v0739eolv38.m701.mp4
+				fns = append(fns[:1], append([]string{fmtIDName}, fns[1:]...)...)
+			} else {
+				// n0687peq62x.p709.mp4 -> n0687peq62x.m709.mp4
+				fns[1] = fmtIDName
+			}
+		} else if len(fns) >= 3 {
+			// delete ID part
+			// e0765r4mwcr.2.mp4 -> e0765r4mwcr.mp4
+			fns = append(fns[:1], fns[2:]...)
+		}
+
 		var urls []downloader.URLData
 		var totalSize int64
 		var filename string
 		for part := 1; part < clips+1; part++ {
 			// Multiple fragments per format
-			if fc > 0 {
-				fns[2] = strconv.Itoa(part)
+			if fmtIDPrefix == "p" {
+				if len(fns) < 4 {
+					// If the number of fragments > 0, the filename needs to add the number of fragments
+					// n0687peq62x.p709.mp4 -> n0687peq62x.p709.1.mp4
+					fns = append(fns[:2], append([]string{strconv.Itoa(part)}, fns[2:]...)...)
+				} else {
+					fns[2] = strconv.Itoa(part)
+				}
 			}
 			filename = strings.Join(fns, ".")
 			html := request.Get(

--- a/extractors/qq_test.go
+++ b/extractors/qq_test.go
@@ -37,7 +37,16 @@ func TestQQ(t *testing.T) {
 				URL:     "https://v.qq.com/iframe/player.html?vid=v0739eolv38",
 				Title:   "PGI国际邀请赛，FPP第四局，OMG强势吃鸡，全场观众高喊OMG",
 				Size:    10714773,
-				Quality: "标清;(270P)",
+				Quality: "高清;(480P)",
+			},
+		},
+		{
+			name: "fmt ID test",
+			args: test.Args{
+				URL:     "https://v.qq.com/x/cover/2aya3ibdmft6vdw/e0765r4mwcr.html",
+				Title:   "《卡路里》出圈！妖娆男子教学广场舞版，大妈表情亮了！",
+				Size:    14112979,
+				Quality: "超清;(720P)",
 			},
 		},
 	}

--- a/request/request.go
+++ b/request/request.go
@@ -93,8 +93,9 @@ func Request(
 		if requestError == nil && res.StatusCode < 400 {
 			break
 		} else if i+1 >= config.RetryTimes {
+			log.Println(url)
 			panic(fmt.Sprintf(
-				"request error: %s HTTP %d", requestError, res.StatusCode,
+				"request error: %v HTTP %d", requestError, res.StatusCode,
 			))
 		}
 		time.Sleep(1 * time.Second)

